src/allotropy/allotrope/schema_mappers/adm/pcr/rec/_2024/_09/qpcr.py

```diff
@@ -0,0 +1,369 @@
+    TQuantityValueUnitless,
+)
+from allotropy.allotrope.models.shared.definitions.definitions import (
+    InvalidJsonFloat,
+    TDatacube,
+)
+from allotropy.allotrope.schema_mappers.data_cube import DataCube, get_data_cube
+from allotropy.allotrope.schema_mappers.schema_mapper import SchemaMapper
+from allotropy.constants import ASM_CONVERTER_VERSION
+from allotropy.parsers.utils.values import assert_not_none, quantity_or_none
+
+
+@dataclass
+class DataSource:
+    identifier: str
+    feature: str
+
+
+@dataclass
+class Error:
+    error: str
+    feature: str
+
+
+@dataclass
+class CalculatedDataItem:
+    identifier: str
+    name: str
+    value: float
+    unit: str
+    data_sources: list[DataSource]
+
+
+@dataclass
+class CalculatedData:
+    items: list[CalculatedDataItem]
+
+
+@dataclass
+class ProcessedData:
+    # Settings
+    cycle_threshold_value_setting: float
+    cycle_threshold_result: float
+    automatic_cycle_threshold_enabled_setting: bool | None = None
+    automatic_baseline_determination_enabled_setting: bool | None = None
+    baseline_determination_start_cycle_setting: int | None = None
+    baseline_determination_end_cycle_setting: int | None = None
+    genotyping_determination_method_setting: float | None = None
+
+    # Results
+    genotyping_determination_result: str | None = None
+    normalized_reporter_result: float | None = None
+    baseline_corrected_reporter_result: float | None = None
+    normalized_reporter_data_cube: DataCube | None = None
+    baseline_corrected_reporter_data_cube: DataCube | None = None
+
+    # Metadata
+    comments: str | None = None
+
+    custom_info: dict[str, Any] | None = None
+
+
+@dataclass
+class Measurement:
+    # Measurement metadata
+    identifier: str
+    timestamp: str
+    sample_identifier: str
+    target_identifier: str
+    location_identifier: str
+    group_identifier: str | None = None
+
+    # Settings
+    pcr_detection_chemistry: str | None = None
+
+    # Optional measurement metadata
+    sample_role_type: str | None = None
+    well_location_identifier: str | None = None
+    well_plate_identifier: str | None = None
+
+    # Optional settings
+    total_cycle_number_setting: float | None = None
+    reporter_dye_setting: str | None = None
+    quencher_dye_setting: str | None = None
+    passive_reference_dye_setting: str | None = None
+
+    # Processed data
+    processed_data: ProcessedData | None = None
+    reporter_dye_data_cube: DataCube | None = None
+    passive_reference_dye_data_cube: DataCube | None = None
+    melting_curve_data_cube: DataCube | None = None
+
+    # Custom metadata
+    custom_info: dict[str, Any] | None = None
+    sample_custom_info: dict[str, Any] | None = None
+    device_control_custom_info: dict[str, Any] | None = None
+
+    # Error document
+    error_document: list[Error] | None = None
+
+
+@dataclass
+class MeasurementGroup:
+    measurements: list[Measurement]
+    plate_well_count: int | InvalidJsonFloat
```

Comment:
plate_well_count can be None now
```suggestion
    plate_well_count: int | None
```